File: 5/5.go
package main

import (
	"fmt"
)

func main() {
	var n int64 = 1
	for {
		if divisibleByAll(n, 20) {
			fmt.Println(n)
			break
		}
		n++
	}
}

func divisibleByAll(n int64, upto int64) bool {
	if n < upto {
		return false
	}
	var i int64
	for i = 1; i <= upto; i++ {
		if n%i != 0 {
			return false
		}
	}
	return true
}

File: 5/5.rs
fn main() {
    let mut n: u64 = 1;
    loop {
        if divisible_by_all(n, 20) {
            break;
        }
        n+=1;
    }
    println!("{:u}", n);
}

fn divisible_by_all(n: u64, upto: u64) -> bool {
    if n < upto {
        return false;
    }
    for i in range(1, upto+1) {
        if n%i != 0 {
            return false;
        }
    }
    return true;
}
